package/MDAnalysis/analysis/base.py

```diff
@@ -351,46 +395,274 @@ def _setup_frames(self, trajectory, start=None, stop=None, step=None,
+        Set things up before the analysis loop begins.
+        Note that `self.results` is initialized already in :meth:`self.__init__`
+        with an empty instance of :class:`MDAnalysis.analysis.results.Results`
+        object. You can still call your attributes as if they were usual ones,
+        `Results` just keeps track of that to be able to run a proper aggregation
+        after a parallel run, if necessary.
+        """
         pass  # pylint: disable=unnecessary-pass
 
     def _conclude(self):
         """Finalize the results you've gathered.
 
         Called at the end of the :meth:`run` method to finish everything up.
+        Note that aggregation of results from individual workers happens
+        in :meth:`self.run()`, so here you have to implement everything as if
+        you had a non-parallel run. If you want to enable proper aggregation
+        for parallel runs for you analysis class, implement `self._get_aggregator`
+        and check :mod:`MDAnalysis.analysis.results` for how to use it.
         """
         pass  # pylint: disable=unnecessary-pass
 
-    def run(self, start=None, stop=None, step=None, frames=None,
-            verbose=None, *, progressbar_kwargs={}):
+    def _compute(self, indexed_frames: np.ndarray,
+                 verbose: bool = None,
+                 *, progressbar_kwargs={}) -> "AnalysisBase":
+        """Perform the calculation on a balanced slice of frames
+        that have been setup prior to that using _setup_computation_groups()
+
+        Parameters
+        ----------
+        indexed_frames : np.ndarray
+            np.ndarray of (n, 2) shape, where first column is frame iteration indices
+            and second is frame numbers
+
+        verbose : bool, optional
+            Turn on verbosity
+
+        progressbar_kwargs : dict, optional
+            ProgressBar keywords with custom parameters regarding progress bar
+            position, etc; see :class:`MDAnalysis.lib.log.ProgressBar`
+            for full list.
+
+        .. versionadded:: 2.8.0
+        """
+        logger.info("Choosing frames to analyze")
+        # if verbose unchanged, use class default
+        verbose = getattr(self, "_verbose", False) if verbose is None else verbose
+
+        frames = indexed_frames[:, 1]
+
+        logger.info("Starting preparation")
+        self._prepare_sliced_trajectory(slicer=frames)
+        self._prepare()
+        if len(frames) == 0:  # if `frames` were empty in `run` or `stop=0`
+            return self
+
+        for idx, ts in enumerate(ProgressBar(
+                self._sliced_trajectory,
+                verbose=verbose,
+                **progressbar_kwargs)):
+            self._frame_index = idx  # accessed later by subclasses
+            self._ts = ts
+            self.frames[idx] = ts.frame
+            self.times[idx] = ts.time
+            self._single_frame()
+        logger.info("Finishing up")
+        return self
+
+    def _setup_computation_groups(
```

Comment:
Is this the right word? "computation" reads somewhat incorrectly here

Reply:
It's certainly the word I was looking for, as in "something to be computed". If you believe some other word should be used, please correct me!